File: LeetCode75/array_and_string/increasing_triplet_subsequence/main.py
from typing import List

class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        small = float('inf')
        mid = small

        for num in nums:
            if num < small:
                small = num
            elif num < mid and num > small:
                mid = num
            elif num > mid:
                return True
            
        return False

if __name__ == "__main__":
    nums = [1, 1, -2, 6]
    solution = Solution()
    print(solution.increasingTriplet(nums))

File: LeetCode75/array_and_string/increasing_triplet_subsequence/main.c
#include <stdio.h>
#include <stdbool.h>
#include <limits.h>

bool increasingTriplet(int* nums, int numsSize)
{
    if (!nums || numsSize <= 0)
        return false;

    int small = INT_MAX;
    int mid = INT_MAX;

    for (int i = 0; i < numsSize; i++)
    {
        if (nums[i] < small)
            small = nums[i];
        else if (nums[i] < mid && nums[i] > small)
            mid = nums[i];
        else if (nums[i] > mid)
            return true;
    }

    return false;
}

int main(void)
{
    int nums[] = {1,1,-2,6};
    int size = (int)(sizeof(nums) / sizeof(nums[0]));
    printf("Result: %d\n", increasingTriplet(nums, size));
}